server.py

```diff
@@ -173,19 +187,114 @@ def display_countries():
     return jsonify(country_information)
 
 
-@app.route('/signin')
-def sign_in():
-    pass
+@app.route('/login', methods=['GET'])
+def login_form():
+    """Show form for user login"""
+
+    return render_template("login_form.html")
+
+
+@app.route('/login', methods=['POST'])
+def login_process():
+    """Process login"""
+
+    email = request.form["email"]
```

Comment:
In order to decrease the chance of name space collisions (https://en.wikipedia.org/wiki/Naming_collision), it's considered best practice to give the variables names that indicate where they came from. Since you're filtering by `email` below, it might be a good idea to name this `form_email` or something like that

Reply:
Same with `password1